ui-demo/src/main/java/de/cotto/lndmanagej/ui/demo/data/DeriveDataUtil.java

```diff
@@ -81,30 +81,32 @@ static OnChainCostsDto deriveOnChainCosts(ChannelId channelId) {
     }
 
     static OpenInitiator deriveOpenInitiator(ChannelId channelId) {
-        boolean local = createRandomGenerator(channelId).nextInt(10) <= 3;
+        boolean local = createRandomGenerator(channelId.getShortChannelId()).nextInt(10) <= 3;
         return local ? LOCAL : REMOTE;
     }
 
     static PoliciesForLocalChannel derivePolicies(ChannelId channelId) {
-        return new PoliciesForLocalChannel(derivePolicy(channelId), derivePolicy(channelId));
+        Policy local = derivePolicy(channelId.getShortChannelId() + 1);
+        Policy remote = derivePolicy(channelId.getShortChannelId() + 2);
+        return new PoliciesForLocalChannel(local, remote);
     }
 
-    static Policy derivePolicy(ChannelId channelId) {
-        RandomGenerator rand = createRandomGenerator(channelId);
+    static Policy derivePolicy(long randomGeneratorSeed) {
+        RandomGenerator rand = createRandomGenerator(randomGeneratorSeed);
         long feeRate = rand.nextLong(100) * 10;
         Coins baseFee = Coins.ofMilliSatoshis(rand.nextLong(2) * 1000);
-        boolean enabled = rand.nextInt(10) == 0;
+        boolean enabled = rand.nextInt(10) != 0;
         int timeLockDelta = (rand.nextInt(5) + 1) * 10;
         return new Policy(feeRate, baseFee, enabled, timeLockDelta, MAX_HTLC);
     }
 
     static ChannelStatusDto deriveChannelStatus(ChannelId channelId) {
-        boolean privateChannel = createRandomGenerator(channelId).nextInt(10) <= 1;
+        boolean privateChannel = createRandomGenerator(channelId.getShortChannelId()).nextInt(10) <= 1;
         return ChannelStatusDto.createFromModel(new ChannelStatus(privateChannel, true, false, OPEN));
     }
 
     static long deriveRating(ChannelId channelId) {
-        return createRandomGenerator(channelId).nextInt(100_000);
+        return createRandomGenerator(channelId.getShortChannelId()).nextInt(100_000);
```

Comment:
Instead of fixing every invocation, you could just have a second method `createRandomGenerator` that (still) accepts `ChannelId`.